src/compiler/config/outputs/validate-www.ts

```diff
@@ -34,105 +33,151 @@ export const validateWww = (config: d.ValidatedConfig, diagnostics: d.Diagnostic
   }
 
-  return userWwwOutputs.reduce(
-    (
-      outputs: (d.OutputTargetWww | d.OutputTargetDistLazy | d.OutputTargetCopy | d.OutputTargetDistGlobalStyles)[],
-      o,
-    ) => {
-      const outputTarget = validateWwwOutputTarget(config, o, diagnostics);
-      outputs.push(outputTarget);
-
-      // Add dist-lazy output target
-      const buildDir = outputTarget.buildDir;
-      outputs.push({
-        type: DIST_LAZY,
-        dir: buildDir,
-        esmDir: buildDir,
-        systemDir: config.buildEs5 ? buildDir : undefined,
-        systemLoaderFile: config.buildEs5 ? join(buildDir, `${config.fsNamespace}.js`) : undefined,
-        polyfills: outputTarget.polyfills,
-        isBrowserBuild: true,
-      });
-
-      // Copy for dist
-      outputs.push({
-        type: COPY,
-        dir: buildDir,
-        copyAssets: 'dist',
-      });
-
-      // Copy for www
-      outputs.push({
-        type: COPY,
-        dir: outputTarget.appDir,
-        copy: validateCopy(outputTarget.copy, [
-          { src: 'assets', warn: false },
-          { src: 'manifest.json', warn: false },
-        ]),
-      });
-
-      // Generate global style with original name
-      outputs.push({
-        type: DIST_GLOBAL_STYLES,
-        file: join(buildDir, `${config.fsNamespace}.css`),
-      });
-
-      return outputs;
-    },
-    [],
-  );
+  const outputs: (
+    | d.ValidatedOutputTargetWww
+    | d.OutputTargetDistLazy
+    | d.OutputTargetCopy
+    | d.OutputTargetDistGlobalStyles
+  )[] = [];
+
+  for (const userOutputTarget of userWwwOutputs) {
+    const outputTarget = validateWwwOutputTarget(config, userOutputTarget, diagnostics);
+    outputs.push(outputTarget);
+
+    // Add dist-lazy output target
+    const buildDir = outputTarget.buildDir;
+    outputs.push({
+      type: DIST_LAZY,
+      dir: buildDir,
+      esmDir: buildDir,
+      systemDir: config.buildEs5 ? buildDir : undefined,
+      systemLoaderFile: config.buildEs5 ? join(buildDir, `${config.fsNamespace}.js`) : undefined,
+      polyfills: outputTarget.polyfills,
+      isBrowserBuild: true,
+    });
+
+    // Copy for dist
+    outputs.push({
+      type: COPY,
+      dir: buildDir,
+      copyAssets: 'dist',
+    });
+
+    // Copy for www
+    outputs.push({
+      type: COPY,
+      dir: outputTarget.appDir,
+      copy: validateCopy(outputTarget.copy, [
+        { src: 'assets', warn: false },
+        { src: 'manifest.json', warn: false },
+      ]),
+    });
+
+    // Generate global style with original name
+    outputs.push({
+      type: DIST_GLOBAL_STYLES,
+      file: join(buildDir, `${config.fsNamespace}.css`),
+    });
+  }
+
+  return outputs;
 };
 
 const validateWwwOutputTarget = (
   config: d.ValidatedConfig,
   outputTarget: d.OutputTargetWww,
   diagnostics: d.Diagnostic[],
-) => {
-  if (!isString(outputTarget.baseUrl)) {
-    outputTarget.baseUrl = '/';
+): d.ValidatedOutputTargetWww => {
```

Comment:
Broke a lot of these checks into separate helpers since certain fields often relied on each other. Didn't document the functions since that seems more like documentation for the sake of documentation rather than documentation that would actually add extra context

Reply:
I think it's pretty clear what `validateAppDir` does!